test/test_embed1p.jl

```diff
@@ -0,0 +1,51 @@
+
+##
+
+using ACE, ACEatoms, JuLIP, LinearAlgebra, ACEbase, Test
+using ACEatoms: Species1PBasis, ZμRnYlm_1pbasis, AtomState,
+                rand_ACEConfig, PopZμRnYlm_1pbasis, rand_ACEConfig_pop
+using ACEbase.Testing
+using ACE: evaluate, evaluate_d, State 
+using StaticArrays
+
+using ACEatoms: EmbedCat1pBasis
+
+
+##
+
+categories = [:O, :H, :C, :N] 
+embeddings = Dict( :O => randn(3), :H => randn(3), :C => randn(3), :N => randn(3) )
```

Comment:
Here I'm deliberately choosing `(Sk)_k = 1..3` which i.e. the embedding vector is shorter than the number of categories. If we were to choose each vector of length 4 then the size of this basis would be the same as the `ZmuRnYlm` basis and we gain nothing I think?